test/TierRegistry/TierRegistry.t.sol

```diff
@@ -0,0 +1,775 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+import "forge-std/Test.sol";
+import {ITierRegistry} from "../../src/TierRegistry.sol";
+import {TransferHelper} from "../../src/libraries/TransferHelper.sol";
+import {TierRegistryTestSuite} from "./TierRegistryTestSuite.sol";
+import "openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract TierRegistryTest is TierRegistryTestSuite {
+    using SafeERC20 for IERC20;
+
+    event PurchasedTier(uint256 indexed fid, uint256 indexed tier, uint256 forDays, address payer);
+    event DeactivateTier(uint256 indexed tier);
+    event SetTier(
+        uint256 indexed tier,
+        uint256 minDays,
+        uint256 maxDays,
+        address vault,
+        address paymentToken,
+        uint256 tokenPricePerDay
+    );
+
+    function testVersion() public {
+        assertEq(tierRegistry.VERSION(), "2025.06.01");
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            PURCHASE TIER 
+    //////////////////////////////////////////////////////////////*/
+
+    function testFuzzPurchaseTier(uint256 fid, uint64 price, uint64 forDays, address payer, address vault) public {
+        vm.assume(payer != address(0));
+        vm.assume(price != 0);
+        vm.assume(vault != address(0));
+        vm.assume(forDays != 0);
+        uint256 tier = _addTier(address(token), price, DEFAULT_MIN_DAYS, DEFAULT_MAX_DAYS, vault);
+        _purchaseTier(fid, tier, forDays, payer);
+    }
+
+    function testFuzzPurchaseTierWithNoTime(uint256 fid, uint64 price, address payer, address vault) public {
+        vm.assume(payer != address(0));
+        vm.assume(price != 0);
+        vm.assume(vault != address(0));
+        uint256 tier = _addTier(address(token), price, DEFAULT_MIN_DAYS, DEFAULT_MAX_DAYS, vault);
+        vm.expectRevert(ITierRegistry.InvalidDuration.selector);
+        tierRegistry.purchaseTier(fid, tier, 0);
+    }
+
+    function testFuzzPurchaseTierWhenPaused(
+        uint256 fid,
+        uint64 price,
+        uint64 forDays,
+        address payer,
+        address vault
+    ) public {
+        vm.assume(payer != address(0));
+        vm.assume(vault != address(0));
+        vm.assume(price != 0);
+        vm.assume(forDays != 0);
```

Comment:
`vm.assume` throws away the test run if the fuzzer selects an invalid value. if you add many `vm.assume` statements, you may start bumping into a "max runs rejected" error. 

when there's a known boundary on the input that you don't want to fuzz over, it's better to use the `bound` helper, which constrains the input to a given range. (in this case we can probably use `bound` for "assume valid tier parameters")

(I'll go through and update, not a blocker!)